axum-core/src/extract/request_parts.rs

```diff
@@ -106,9 +106,7 @@ where
                 }
             })?;
 
-        let string = std::str::from_utf8(&bytes)
-            .map_err(InvalidUtf8::from_err)?
-            .to_owned();
+        let string = String::from_utf8(bytes.into()).map_err(InvalidUtf8::from_err)?;
```

Comment:
Won't the `bytes.into()` still reallocate the data? Unless there is some special code in the vtable of whatever representation of `Bytes` that underlies this that checks the reference count and avoids a copy if it is 1?

Reply:
Is that what this does? https://docs.rs/bytes/1.6.1/src/bytes/bytes.rs.html#1108-1128

Reply:
From what I remember when I looked at `bytes`, you have two ways `Bytes` may be represented. Either with an owned `Vec` or if there are at least two instances referencing the same allocation (even if disjunct), it gets promoted to an `Arc`-backed storage.

But I think the bytes will already be cloned when the request body is collected so I think it should prevent one clone from happening.